nodejs/week3/meal_sharing_app/src/backend/api/meals.js

```diff
@@ -0,0 +1,142 @@
+const express = require("express");
+const app = express();
+const router = express.Router();
+const knex = require("../database");
+
+app.use(express.json());
+app.use(express.urlencoded({
+  extended: true
+}));
+
+//GET api/meals/ query parameters
+
+router.get("/", async function (req, res) {
+  const {
+    maxPrice,
+    title,
+    availableReservations,
+    createdAfter,
+    limit
+  } = req.query;
+  const mealTitles = await knex.select("*").table("meal");
+
+  if (Object.keys(req.query).length === 0) {
```

Comment:
Lets help the developers reading through this code a little:
```suggestion
  const requestHasNoQueryParameters = Object.keys(req.query).length === 0;
  if (requestHasNoQueryParameters ) {
```